dask/base.py

```diff
@@ -0,0 +1,85 @@
+import warnings
+from operator import attrgetter
+
+from toolz import merge, groupby, unique
+
+from .context import _globals
+
+
+class Base(object):
+    """Base class for dask collections"""
+
+    def visualize(self, optimize_graph=False):
+        from dask.dot import dot_graph
+        if optimize_graph:
+            return dot_graph(self._config.optimize(self.dask, self._keys()))
+        else:
+            return dot_graph(self.dask)
+
+    def _visualize(self, optimize_graph=False):
+        warn = DeprecationWarning("``_visualize`` is deprecated, use "
+                                  "``visualize`` instead.")
+        warnings.warn(warn)
+        return self.visualize(optimize_graph)
+
+    def compute(self, **kwargs):
+        return compute(self, **kwargs)[0]
+
+
+class Config(object):
+    """Specifies a configuration for a dask collection.
+
+    This holds functions that are used to specialize dask schedulers for a
+    specific collection.
+
+    Parameters
+    ----------
+    optimize : callable
+        Takes a dask and iterable of keys, and returns an optimized dask.
+    default_get : callable
+        The default scheduler to use (e.g. ``dask.threaded.get``)
+    finalize : callable
+        Takes a dask collection and an iterable of results, and converts them
+        into the desired python type.
+    """
+    def __init__(self, optimize, default_get, finalize):
+        self.optimize = optimize
+        self.default_get = default_get
+        self.finalize = finalize
```

Comment:
I'm tempted to leave these on the collection class itself rather than bundling them into a `Config` class.  I've found some benefit to maintaining a very flat class structure.  Thoughts?